testing/sdk_marathon.py

```diff
@@ -174,21 +161,22 @@ def install_app(app_definition: dict, timeout=TIMEOUT_SECONDS) -> None:
     log.info("Installing app: {}".format(app_name))
 
     @retrying.retry(stop_max_delay=timeout * 1000, wait_fixed=2000)
-    def _install():
+    def _install() -> MarathonDeploymentResponse:
         response = sdk_cmd.cluster_request(
             "POST", _api_url("apps"), json=app_definition, log_args=False, raise_on_error=False
         )
-        return _handle_marathon_deployment_response(response)
+        try:
+            deployment_response = MarathonDeploymentResponse(response)
+        except requests.HTTPError as e:
+            log.error("HTTP error response code while installing: %s", response.json()["message"])
```

Comment:
Should we really call `response.json()` here?

Why not have `MarathonDeploymentResponse` wrap an instance `requests.Response`. We could then encapsulate the logic in this error handling in methods of the class.

Something like:
```python
deployment_response = MarathonDeploymentResponse(response)
if deployment_response.app_exists():
    destroy_app(app_name, timeout)
deployment_response.raise_for_status()  # This should explicitly ignore the 409 case that was already checked.
return deployment_response
```

Reply:
(ok, this breaks the other usages of this response a little, but we could then include similar checks in these two places)

Reply:
I'd rather keep `MarathonDeploymentResponse` as small as possible. Handling different response codes seems pretty dependent on the endpoint. It's not necessary to call `response.json()` here again because it was already done during `MarathonDeploymentResponse` init. I'll remove it.